gpflow/sgpr.py

```diff
@@ -121,21 +121,20 @@ def build_likelihood(self):
         SGPR notebook.
         """
 
-        num_inducing = tf.shape(self.Z)[0]
         num_data = tf.cast(tf.shape(self.Y)[0], settings.dtypes.float_type)
         output_dim = tf.cast(tf.shape(self.Y)[1], settings.dtypes.float_type)
 
         err = self.Y - self.mean_function(self.X)
         Kdiag = self.kern.Kdiag(self.X)
-        Kuf = self.kern.K(self.Z, self.X)
-        Kuu = self.kern.K(self.Z) + tf.eye(num_inducing, dtype=float_type) * settings.numerics.jitter_level
+        Kuf = self.feat.Kuf(self.kern, self.X)
+        Kuu = self.feat.Kuu(self.kern)
         L = tf.cholesky(Kuu)
         sigma = tf.sqrt(self.likelihood.variance)
 
         # Compute intermediate matrices
         A = tf.matrix_triangular_solve(L, Kuf, lower=True) / sigma
         AAT = tf.matmul(A, A, transpose_b=True)
-        B = AAT + tf.eye(num_inducing, dtype=float_type)
+        B = AAT + tf.eye(self.num_inducing, dtype=float_type)
```

Comment:
Similarly, here, the number of inducing points can also change after object construction.